modules/core/02-client/keeper/keeper.go

```diff
@@ -493,3 +486,25 @@ func (k *Keeper) ScheduleIBCSoftwareUpgrade(ctx sdk.Context, plan upgradetypes.P
 
 	return nil
 }
+
+// getLightClientModule checks that the clientType of clientID is allowed and returns the corresponding light client module
+func (k *Keeper) getLightClientModule(ctx sdk.Context, clientID string) (exported.LightClientModule, error) {
+	clientType, _, err := types.ParseClientIdentifier(clientID)
+	if err != nil {
+		return nil, errorsmod.Wrapf(err, "unable to parse client identifier %s", clientID)
+	}
+
+	if !k.GetParams(ctx).IsAllowedClient(clientType) {
+		return nil, errorsmod.Wrapf(
+			types.ErrInvalidClientType,
+			"client (%s) type %s is not in the allowed client list", clientID, clientType,
+		)
+	}
+
+	clientModule, found := k.router.GetRoute(clientID)
+	if !found {
+		return nil, errorsmod.Wrap(types.ErrRouteNotFound, clientID)
+	}
```

Comment:
I feel like this params check could be moved into `router.GetRoute()`. 

We already parse the clientID in there, so the changes would be to accept a context and return and error rather than a boolean. I'd be in favour of doing this in a following PR. I can create an issue if others are onboard